bin/prefill_changelog

```diff
@@ -11,47 +17,49 @@ if [[ "$1" == "--dry-run" ]]; then
   shift
 fi
 
-desired_version=$(echo ${1} | sed 's/^v//')
-tags=$(git tag | grep '^v' | sed 's/^v//')
-latest_tag=$(echo "$tags" | sort -rV | head -1)
+desired_tag=${1//v/}
+latest_tag=$(git tag | sed 's/^v//' | most_recent_version)
+latest_version="v${latest_tag}"
+desired_version="v${desired_tag}"
 
-echo "Latest known tag: 'v${latest_tag}'"
-echo "New version: 'v${desired_version}'"
+echo "Latest known tag: '${latest_tag}'"
+echo "New version: 'v${desired_tag}'"
 
 # Sanity check
-newer_version=$(echo -e "${latest_tag}\n${desired_version}" | sort -rV | head -1)
-if [[ "${newer_version}" != "${desired_version}" ]]; then
-  echo "ERROR! Desired version 'v${desired_version}' is below latest version 'v${latest_tag}'!"
+newest_tag=$(printf '%s\n%s' "${latest_tag}" "${desired_tag}" | most_recent_version)
+if [[ "${newest_tag}" != "${desired_tag}" ]]; then
+  echo "ERROR! Desired version '${desired_version}' is below latest version '${latest_version}'!"
   echo "Exiting!"
   exit 1
 fi
 
 echo "Fetching git commit data..."
-git_diff=$(git log --format="- %s" "v${latest_tag}..HEAD")
+git_diff=$(git log --format="- %s" "${latest_version}..HEAD")
 change_date=$(date +%Y-%m-%d)
 
-changelog_snippet="\n"
-changelog_snippet+="## [${desired_version}] ${change_date}\n\n"
-changelog_snippet+="### TODO: Don't commit this without proofreading!!!\n\n"
-changelog_snippet+="### Added\n\n"
-changelog_snippet+="### Changed\n\n"
+changelog_snippet='\n'
+changelog_snippet+="## [${desired_tag}] ${change_date}"'\n\n'
+changelog_snippet+="### TODO: Don't commit this without proofreading!!!"'\n\n'
```

Comment:
I think the single quotes are causing this text to be parsed incorrectly - running the script results in output that looks like this:
```
\n## [0.6.5] 2019-03-18\n\n### TODO: Don't commit this without proofreading!!!\n\n### Added\n\n### Changed\n\n- Address a couple of Geri's comments
```

Reply:
I rewrote this and it worked for me, you should re-run the test yourself though to be 100% sure.

What happened was I forgot that in my refactor `changlog_snippet` was supposed to be the printf format string, not the data string.  I updated it to reflect that.